### PR TITLE
Add `load_from_json` and `save_to_json`  to `SingleTableMetadata`

### LINKED ISSUE #874: Add load_from_json and save_to_json methods to SingleTableMetadata
### Problem Description

As a user, it would be useful if I could create an instance of the `SingleTableMetadata` class from a json file. It would also be useful if I could save an existing instance to a json file.

### Expected behavior

- Add a `load_from_json` method
    - Parameters:
        - `filepath`: string that is the path to json file
    - Errors:
        - If the json file doesn't contain the `SCHEMA_VERSION` attribute, then raise the following error
        ```python
        Error: This metadata file is incompatible with the SingleTableMetadata class and version.
        ``` 
```python
>>> metadata = SingleTableMetadata.load_from_json(filepath='project/metadata.json')
```
- Add a `save_to_json` method
    - Add `SCHEMA_VERSION` to dict privately
    - Parameters:
        - `filepath`: string that is the path to save the file to
    - Errors:
        - If there is already a file in the path location, raise the following error
        ```python
        Error: A file named 'metadata.json' already exists in this folder. Please specify a different filename.
        ```
```python
>>> metadata.save_to_json(filepath='project/metadata.json')
```

### PR DESCRIPTION
**Pull request by pvk-developer**
Resolves #874

### CODE REVIEW

File: tests/unit/metadata/test_single_table.py [hunk 1]
```diff
@@ -105,6 +109,183 @@ def test__load_from_dict(self):
...
+            - String representing a filepath.
+
+        Side Effects:
+            - A ``ValueError`` is raised pointing that the ``file`` does not exist.
+        """
+        # Setup
+        mock_path.return_value.exists.return_value = False
+        mock_path.return_value.name = 'filepath.json'
+
+        # Run
```
Comment (katxiao) [inline tests/unit/metadata/test_single_table.py:131]: small nit: I have seen the comment `# Run and Assert` in cases like these.

**Review by katxiao — APPROVED** [review]
LGTM!

File: tests/unit/metadata/test_single_table.py (hunk 1 above)
Comment (pvk-developer) [reply tests/unit/metadata/test_single_table.py:131]: Nice catch, I usually write `# Run / Assert` but this time I missed it. WIll address it asap.

**Review by amontanez24 — CHANGES_REQUESTED** [review]
One thing concerning constraints, but the rest looks good!

**Pull request merged**